experimental/ir/ir_value.go

```diff
@@ -0,0 +1,479 @@
+// Copyright 2020-2025 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ir
+
+import (
+	"math"
+
+	"github.com/bufbuild/protocompile/experimental/ast"
+	"github.com/bufbuild/protocompile/experimental/ast/predeclared"
+	"github.com/bufbuild/protocompile/experimental/internal"
+	"github.com/bufbuild/protocompile/experimental/ir/presence"
+	"github.com/bufbuild/protocompile/experimental/seq"
+	"github.com/bufbuild/protocompile/internal/arena"
+	"github.com/bufbuild/protocompile/internal/ext/mapsx"
+	"github.com/bufbuild/protocompile/internal/ext/slicesx"
+	"github.com/bufbuild/protocompile/internal/intern"
+)
+
+// This silences unused function lints.
+//
+// These functions will be used by options lowering, which is not yet
+// implemented, but are included here now because they are an essential part
+// of the data structures in this file.
+//
+// They are all together here so it's easier to remember to delete them at the
+// same time, instead of putting //nolint on each function.
```

Comment:
The value of using `nolint` is that when the function becomes used, the linter will require that you remove the nolint comment. With this, we could easily forget to address the TODO and have this left in for a while after things are implemented...

Reply:
OHHHHH right i forgot lmao